File: VB/MyRadioGroup/ExtendedEditor/MyRepositoryItemRadioGroup.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Drawing

Namespace EditorDescendant
	<System.ComponentModel.DesignerCategory("")> _
	Public Class MyRepositoryItemRadioGroup
		Inherits RepositoryItemRadioGroup
		#Region "Service"

		Shared Sub New()
			Register()
		End Sub

		Public Sub New()
			MinItemHeight = 20
			AllowHtmlString = True
		End Sub
		Friend Const EditorName As String = "MyRadioGroup"

		Public Shared Sub Register()
			EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(EditorName, GetType(MyRadioGroup), GetType(MyRepositoryItemRadioGroup), GetType(MyRadioGroupViewInfo), New RadioGroupPainter(), True, EditImageIndexes.RadioGroup, GetType(DevExpress.Accessibility.BaseEditAccessible)))
		End Sub

		Public Overrides ReadOnly Property EditorTypeName() As String

			Get
				Return EditorName
			End Get

		End Property

		#End Region



		Private privateMinItemHeight As Integer
		Public Property MinItemHeight() As Integer
			Get
				Return privateMinItemHeight
			End Get
			Set(ByVal value As Integer)
				privateMinItemHeight = value
			End Set
		End Property


		Private privateAllowHtmlString As Boolean
		Public Property AllowHtmlString() As Boolean
			Get
				Return privateAllowHtmlString
			End Get
			Set(ByVal value As Boolean)
				privateAllowHtmlString = value
			End Set
		End Property


	End Class
End Namespace

File: CS/MyRadioGroup/ExtendedEditor/MyRepositoryItemRadioGroup.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Drawing;

namespace EditorDescendant
{
    [System.ComponentModel.DesignerCategory("")]
    public class MyRepositoryItemRadioGroup : RepositoryItemRadioGroup
    {
        #region Service

        static MyRepositoryItemRadioGroup()
        {
            Register();
        }

        public MyRepositoryItemRadioGroup()
        {
            MinItemHeight = 20;
            AllowHtmlString = true;
        }
        internal const string EditorName = "MyRadioGroup";

        public static void Register()
        {
            EditorRegistrationInfo.Default.Editors.Add(
                new EditorClassInfo(EditorName, typeof(MyRadioGroup),
                typeof(MyRepositoryItemRadioGroup),
                typeof(MyRadioGroupViewInfo),
                new RadioGroupPainter(), true,
                EditImageIndexes.RadioGroup, typeof(DevExpress.Accessibility.BaseEditAccessible)));
        }

        public override string EditorTypeName
        {

            get { return EditorName; }

        }

        #endregion



        public int MinItemHeight { get; set; }


        public bool AllowHtmlString { get; set; }
        
        
    }
}
